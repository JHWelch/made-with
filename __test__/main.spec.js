// @vitest-environment jsdom
import { beforeEach, expect, it } from 'vitest'

beforeEach(async () => {
    const div = document.createElement('div');
    div.id = 'made-with';
    document.body.appendChild(div);

    await import('../main.js');
});


it('should render the made-with component', () => {
    expect(document.getElementById('made-with')).not.toBeNull();
    expect(document.getElementById('spinner')).not.toBeNull();
    expect(document.getElementById('emojis')).not.toBeNull();

    expect(document).toMatchSnapshot();
});
