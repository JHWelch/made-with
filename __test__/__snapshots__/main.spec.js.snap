// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render the made-with component 1`] = `
Document {
  "defaultView": [Object],
  "location": Location {
    "assign": [Function],
    "hash": "",
    "host": "localhost:3000",
    "hostname": "localhost",
    "href": "http://localhost:3000/",
    "origin": "http://localhost:3000",
    "pathname": "/",
    "port": "3000",
    "protocol": "http:",
    "reload": [Function],
    "replace": [Function],
    "search": "",
    "toString": [Function],
  },
}
`;
